### PR TITLE
Liviu popa/sm tm e1e56 crud all categories and subcategories

### PR DESCRIPTION
**Pull request by livpopa**
Added Base and Tests Class

### CODE REVIEW

File: src/test/java/crudAllCaterogiesAndSubcategoriesTests/Base.java [hunk 1]
```diff
@@ -0,0 +1,4 @@
+package crudAllCaterogiesAndSubcategoriesTests;
```
Comment (95teo) [inline src/test/java/crudAllCaterogiesAndSubcategoriesTests/Base.java:1]: You have a typo in the name of the package. Please change package name _crudAllCaterogiesAndSubcategoriesTests_ -> **crudAllCategoriesAndSubcategoriesTests**. Thanks!

Comment (livpopa) [reply src/test/java/crudAllCaterogiesAndSubcategoriesTests/Base.java]: hi Teo. I believe i did the change. Could you please double check?

File: src/main/java/pages/CategoriesPage.java [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+    @FindBy(id = "category-name")
+    private WebElement categoryNameField;
+    @FindBy(id = "categories-box")
+    private WebElement subcategoryNameField;
+    @FindBy(css = "body > div:nth-child(2) > div:nth-child(1) > div:nth-child(3) > div:nth-child(1) > div:nth-child(1) > div:nth-child(3) > form:nth-child(1) > div:nth-child(2) > button:nth-child(1)")
+    private WebElement createButton;
+    @FindBy(xpath = "//div[contains(text(),'Constructii')]")
+    private WebElement readFirstCategory;
+
+
```
Comment (95teo) [inline src/main/java/pages/CategoriesPage.java:33]: Please remove one blank line. Thanks!

Comment (95teo) [inline src/main/java/pages/CategoriesPage.java:33]: Please remove one blank line. Thanks!

**Review by 95teo — APPROVED** [review]

File: src/main/java/pages/CategoriesPage.java (hunk 2 above)
Comment (95teo) [inline src/main/java/pages/CategoriesPage.java:33]: Please remove one blank line. Thanks!

Comment (95teo) [inline src/main/java/pages/CategoriesPage.java:33]: Please remove one blank line. Thanks!

**Comment by 95teo** [comment]
Don't forget to fix the conflicts. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

File: src/test/java/crudAllCategoriesAndSubcategoriesTests/Tests.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    CategoryDetailsGenerator categoryDetailsGenerator;
+
+    @Test
+    public void A_createRandomCategory() {
+        categoriesPage.login("capnv12@gmail.com", "Capnv12");
+        categoryDetailsGenerator = new CategoryDetailsGenerator();
+        categoryDetails = categoryDetailsGenerator.generateRandomCategoryDetails();
+        categoriesPage.createRandomCategory(categoryDetails);
+ //       assertTrue(categoriesPage.checkLinkPage());
+        driver.quit();
```
Comment (95teo) [inline src/test/java/crudAllCategoriesAndSubcategoriesTests/Tests.java:18]: I was refferring to add the @AfterMethod/@AfterSuite methods like in the other Base test classes from other packages. Not putine this line at test level. Thanks!

**Review by 95teo — CHANGES_REQUESTED** [review]

**Review by 95teo — APPROVED** [review]

File: src/main/java/modelsGenerator/CategoryDetailsGenerator.java [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+public class CategoryDetailsGenerator {
+    CategoryDetails categoryDetails;
+    public CategoryDetails generateRandomCategoryDetails() {
+        categoryDetails = new CategoryDetails();
+        Faker faker = new Faker();
+        categoryDetails.setCategoryName(faker.name().name());
+        categoryDetails.setFirstSubcategoryName(faker.name().name());
+        categoryDetails.setSecondSubcategoryName(faker.name().name());
+        categoryDetails.setThirdSubcategoryName(faker.name().name());
+
```
Comment (ElizaPopa) [inline src/main/java/modelsGenerator/CategoryDetailsGenerator.java:15]: Please delete this blank line. Thank you.

**Review by ElizaPopa — APPROVED** [review]

**Pull request merged**